src/langcheck/metrics/eval_clients/_openai.py

```diff
@@ -245,75 +247,82 @@ def get_float_score(
         if language not in ["en", "ja", "de", "zh"]:
             raise ValueError(f"Unsupported language: {language}")
 
-        fn_call_template = get_template(
-            f"{language}/get_score/function_calling.j2"
-        )
-
         options = list(score_map.keys())
-        fn_call_messages = [
-            fn_call_template.render(
-                {
-                    "metric": metric_name,
-                    "unstructured_assessment": unstructured_assessment,
-                    "options": options,
-                }
-            )
-            if unstructured_assessment
-            else None
-            for unstructured_assessment in unstructured_assessment_result
-        ]
 
-        functions = [
+        class Response(BaseModel):
+            score: Literal[tuple(options)]  # type: ignore
+
+        structured_output_template = get_template(
+            f"{language}/get_score/function_calling.j2"
+        )
+        model_inputs = [
             {
-                "name": "save_assessment",
-                "description": f"Save the assessment of {metric_name}.",
-                "parameters": {
-                    "type": "object",
-                    "properties": {
-                        "assessment": {
-                            "type": "string",
-                            "enum": options,
-                            "description": f"The assessment of {metric_name}.",
-                        },
-                    },
-                    "required": ["assessment"],
-                },
+                "model": "gpt-4o-mini",
+                "messages": [
+                    {
+                        "role": "user",
+                        "content": structured_output_template.render(
+                            metric_name=metric_name,
+                            unstructured_assessment=unstructured_assessment,
+                            options=options,
+                        ),
+                    }
+                ],
+                "response_format": Response,
             }
+            for unstructured_assessment in unstructured_assessment_result
         ]
 
-        config_structured_assessments = {
-            "functions": functions,
-            "function_call": {
-                "name": "save_assessment",
-            },
-            "model": "gpt-3.5-turbo",
-        }
-        config_structured_assessments.update(self._openai_args or {})
+        if self._use_async:
+            # A helper function to call the async API.
+            async def _call_async_api() -> list[Any]:
+                responses = await asyncio.gather(  # type: ignore
```

Comment:
```suggestion
                responses = await asyncio.gather(
```

You might not need this one... please double check!!

Reply:
Thanks for catching! Fixed: [8c8ff2e](https://github.com/citadel-ai/langcheck/pull/186/commits/8c8ff2ee953d0482f7a42af4c5674f2573654e6b)